### PR TITLE
Seller OTP login required

### PR DESCRIPTION
**Pull request by theflyingmachine**

### CODE REVIEW

File: customer/models.py [hunk 1]
```diff
@@ -65,35 +66,46 @@ def create_or_update_reaction(message_id, related_id, route, to_number, sender_n
...
     @staticmethod
-    def get_otp(customer):
+    def get_otp(user, user_type):
         # Cleanup 24 hours old OTP
         time_threshold = datetime.now() - timedelta(hours=24)
         LoginOTP.objects.filter(generated_date__lt=time_threshold).delete()
 
         # Check if OPT exists, and attempt remaining
-        current_otp = LoginOTP.objects.filter(customer=customer).first()
+        if user_type == 'customer':
```
Comment (nikithapk) [inline customer/models.py:82]: combine the query using Q() filter depending on the conditions

File: customer/models.py [hunk 2]
```diff
@@ -65,35 +66,46 @@ def create_or_update_reaction(message_id, related_id, route, to_number, sender_n
...
+            current_otp = LoginOTP.objects.filter(customer=user).first()
+        else:
+            current_otp = LoginOTP.objects.filter(seller=user).first()
+
         if not current_otp:
             # Generate New OPT and return
-            current_otp = LoginOTP(customer=customer,
-                                   otp_password=random.randrange(111111, 999999, 6),
-                                   login_attempt=0)
+            if user_type == 'customer':
```
Comment (nikithapk) [inline customer/models.py:89]: use Q() here as well

File: customer/models.py [hunk 3]
```diff
@@ -65,35 +66,46 @@ def create_or_update_reaction(message_id, related_id, route, to_number, sender_n
...
+
             current_otp.save()
 
             # send OTP Notification
             from register.utils import send_whatsapp_message
             wa_body = WA_OTP_MESSAGE_TEMPLATE
             wa_body[
-                'to'] = f"91{DEV_NUMBER}" if RUN_ENVIRONMENT == 'dev' else f"91{customer.contact}"
-            wa_body['template']['components'][0]['parameters'][0]['text'] = customer.name.title()
+                'to'] = f"91{DEV_NUMBER}" if RUN_ENVIRONMENT == 'dev' else f"91{user.contact}"
```
Comment (nikithapk) [inline customer/models.py:104]: use boolean for dev env

File: register/views.py [hunk 4]
```diff
@@ -946,16 +946,42 @@ def landing(request):
         'is_mobile': is_mobile(request),
     }
     if request.method == "POST":
-        username = request.POST.get("username")
+        tenant = None
+        username = request.POST.get("username", '')
```
Comment (nikithapk) [inline register/views.py:950]: take a look at request.POST.get("username") or ' '

File: customer/models.py (hunk 1 above)
Comment (theflyingmachine) [reply customer/models.py]: Updated

File: customer/models.py (hunk 2 above)
Comment (theflyingmachine) [reply customer/models.py]: Updated

File: customer/models.py (hunk 3 above)
Comment (theflyingmachine) [reply customer/models.py]: Updated

File: register/views.py (hunk 4 above)
Comment (theflyingmachine) [reply register/views.py:950]: Updated, and moved username.lower inside check

**Pull request merged**